src/test/java/io/redis/examples/reactive/StringExample.java

```diff
@@ -0,0 +1,104 @@
+// EXAMPLE: set_tutorial
+package io.redis.examples.reactive;
+
+import io.lettuce.core.*;
+import io.lettuce.core.api.reactive.RedisReactiveCommands;
+import io.lettuce.core.api.StatefulRedisConnection;
+// REMOVE_START
+import org.junit.jupiter.api.Test;
+// REMOVE_END
+import reactor.core.publisher.Mono;
+
+import java.util.*;
+
+// REMOVE_START
+import static org.assertj.core.api.Assertions.assertThat;
+// REMOVE_END
+
+public class StringExample {
+
+    // REMOVE_START
+    @Test
+    // REMOVE_END
+    public void run() {
+        RedisClient redisClient = RedisClient.create("redis://localhost:6379");
+
+        try (StatefulRedisConnection<String, String> connection = redisClient.connect()) {
+            RedisReactiveCommands<String, String> reactiveCommands = connection.reactive();
+
+            // STEP_START set_get
+            Mono<Void> setAndGet = reactiveCommands.set("bike:1", "Deimos").doOnNext(v -> {
+                System.out.println(v); // OK
+                // REMOVE_START
+                assertThat(v).isEqualTo("OK");
+                // REMOVE_END
+            }).flatMap(v -> reactiveCommands.get("bike:1")).doOnNext(res -> {
+                // REMOVE_START
+                assertThat(res).isEqualTo("Deimos");
+                // REMOVE_END
+                System.out.println(res); // Deimos
+            }).then();
+            // STEP_END
```

Comment:
```suggestion
// STEP_START set_get
            // Deimos
            Mono<String> setAndGet = reactiveCommands.set("bike:1", "Deimos")
                    .doOnNext(res-> System.out.println(res)) // OK
                    .flatMap(v -> reactiveCommands.get("bike:1"))
                    .doOnNext(res-> System.out.println(res));  // Deimos

            // REMOVE_START
            String[] output = SysOutCaptor.CaptureSystemOut(() ->
                    StepVerifier.create(setAndGet)
                            .assertNext(v -> assertThat(v).isEqualTo("Deimos"))
                            .expectComplete()
                            .verify()
            );
            assertThat(output).containsSequence("OK", "Deimos");
            // REMOVE_END
            // STEP_END
```
I suggest extracting checks for code examples, as a separate step for readability and ease of maintenance

Reply:
After discussion with @tishun, we decided to keep the original option, which should be easier for nontechnical people to contribute examples!